api/src/test/java/com/epam/pipeline/controller/configuration/ServerlessConfigurationControllerTest.java

```diff
@@ -0,0 +1,79 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.controller.configuration;
+
+import com.epam.pipeline.controller.Result;
+import com.epam.pipeline.manager.configuration.ServerlessConfigurationApiService;
+import com.epam.pipeline.test.web.AbstractControllerTest;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+import static org.mockito.Matchers.eq;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+@WebMvcTest(controllers = ServerlessConfigurationController.class)
+public class ServerlessConfigurationControllerTest extends AbstractControllerTest {
+
+    private static final long ID = 1L;
+    private static final String TEST_CONFIG = "testConfig";
+    private static final String RESULT = "RESULT";
+    private static final String SERVERLESS_URL = SERVLET_PATH + "/serverless";
+    private static final String GENERATE_URL = SERVERLESS_URL + "/url/%d";
+    private static final String RUN_URL = SERVERLESS_URL + "/%d/%s";
+    private static final TypeReference<Result<String>> STRING_TYPE = new TypeReference<Result<String>>() { };
+
+    @Autowired
+    private ServerlessConfigurationApiService mockServerlessConfigurationApiService;
+
+    @Test
+    public void shouldFailGenerateUrlForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(GENERATE_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGenerateUrl() throws Exception {
+        Mockito.doReturn(RESULT).when(mockServerlessConfigurationApiService).generateUrl(ID, TEST_CONFIG);
+
+        final MvcResult mvcResult = performRequest(
+                get(String.format(GENERATE_URL, ID)).param("config", TEST_CONFIG)
+        );
+
+        Mockito.verify(mockServerlessConfigurationApiService).generateUrl(ID, TEST_CONFIG);
+        assertResponse(mvcResult, RESULT, STRING_TYPE);
+    }
+
+    @Test
+    public void shouldFailRunForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(RUN_URL, ID, TEST_CONFIG)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldRun() throws Exception {
+        Mockito.doReturn(RESULT).when(mockServerlessConfigurationApiService)
+                .run(eq(ID), eq(TEST_CONFIG), Mockito.any());
+
+        performRequest(get(String.format(RUN_URL, ID, TEST_CONFIG)));
+
+        Mockito.verify(mockServerlessConfigurationApiService).run(eq(ID), eq(TEST_CONFIG), Mockito.any());
+    }
```

Comment:
Could you please check that POST, GET, PUT and DELETE requests can be made. For example parameterized test can be of help here.